src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ClientErrorResultFilter.cs

```diff
@@ -38,16 +37,15 @@ public void OnResultExecuting(ResultExecutingContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (context.Result is IClientErrorActionResult clientErrorActionResult &&
-                clientErrorActionResult.StatusCode is int statusCode &&
-                _clientErrorFactory.TryGetValue(statusCode, out var factory))
+            if (!(context.Result is IClientErrorActionResult clientError))
             {
-                var result = factory(context);
+                return;
+            }
 
-                _logger.TransformingClientError(context.Result.GetType(), result?.GetType(), statusCode);
+            var result = _clientErrorFactory.GetClientError(context, clientError);
```

Comment:
This should be allowed to return null? It makes sense that we'd want to allow users to write code that decides **not** to map something, in which case we'd use the original error.